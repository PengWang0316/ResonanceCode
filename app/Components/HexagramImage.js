import React from 'react';
import PropTypes from 'prop-types';

import { getHexagramImageClassNamesArray, getHexagramBlackImageClassNamesArray } from '../apis/Util';

const HexagramImage = ({ isBlack, imageNumber, isFirstImage }) => {
  const image = isBlack ?
    getHexagramBlackImageClassNamesArray(imageNumber, isFirstImage) :
    getHexagramImageClassNamesArray(imageNumber, isFirstImage);
  // console.log(isFirstImage);
  // console.log(image);
  return (
    <div className="inlineBlock">
      <div className="image-line"><div className={image[5].side} /><div className={image[5].middle} /><div className={image[5].side} /></div>
      <div className="image-line"><div className={image[4].side} /><div className={image[4].middle} /><div className={image[4].side} /></div>
      <div className="image-line"><div className={image[3].side} /><div className={image[3].middle} /><div className={image[3].side} /></div>
      <div className="image-line"><div className={image[2].side} /><div className={image[2].middle} /><div className={image[2].side} /></div>
      <div className="image-line"><div className={image[1].side} /><div className={image[1].middle} /><div className={image[1].side} /></div>
      <div className="image-line"><div className={image[0].side} /><div className={image[0].middle} /><div className={image[0].side} /></div>
    </div>

  );
};
HexagramImage.propTypes = {
  imageNumber: PropTypes.string.isRequired,
  isFirstImage: PropTypes.bool.isRequired,
  isBlack: PropTypes.bool.isRequired
};
export default HexagramImage;
