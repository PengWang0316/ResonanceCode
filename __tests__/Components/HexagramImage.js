import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import HexagramImage from '../../app/Components/HexagramImage';

// const mockGetHexagramImageClassNamesArray = jest.fn(() => new Array(6).fill({ side: 'side', middle: 'middle' }));
// const mockGetHexagramBlackImageClassNamesArray = jest.fn(() => new Array(5).fill({ side: 'blackSide', middle: 'blackMiddle' }));
jest.mock('../../app/apis/Util', () => ({ getHexagramImageClassNamesArray: jest.fn(() => new Array(6).fill({ side: 'side', middle: 'middle' })), getHexagramBlackImageClassNamesArray: jest.fn(() => new Array(6).fill({ side: 'blackSide', middle: 'blackMiddle' })) }));

describe('HexagramImage test', () => {
  const defaultProps = {
    imageNumber: '1',
    isFirstImage: true,
    isBlack: true
  };
  const getShallowComponent = (props = defaultProps) =>
    shallow(<HexagramImage {...props} />);

  test('isBlack', () => {
    const Util = require('../../app/apis/Util');
    getShallowComponent();
    expect(Util.getHexagramBlackImageClassNamesArray).toHaveBeenLastCalledWith('1', true);
  });

  test('isNotBlack', () => {
    const Util = require('../../app/apis/Util');
    getShallowComponent({
      imageNumber: '2',
      isFirstImage: false,
      isBlack: false
    });
    expect(Util.getHexagramImageClassNamesArray).toHaveBeenLastCalledWith('2', false);
  });

  test('HexagramImage snapshot', () => expect(renderer.create(<HexagramImage {...defaultProps} />).toJSON()).toMatchSnapshot());
});
