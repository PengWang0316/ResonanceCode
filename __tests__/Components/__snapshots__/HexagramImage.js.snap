// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HexagramImage test HexagramImage snapshot 1`] = `
<div
  className="d-inline-block"
>
  <div
    className="imageLine"
  >
    <div
      className="blackSide"
    />
    <div
      className="blackMiddle"
    />
    <div
      className="blackSide"
    />
  </div>
  <div
    className="imageLine"
  >
    <div
      className="blackSide"
    />
    <div
      className="blackMiddle"
    />
    <div
      className="blackSide"
    />
  </div>
  <div
    className="imageLine"
  >
    <div
      className="blackSide"
    />
    <div
      className="blackMiddle"
    />
    <div
      className="blackSide"
    />
  </div>
  <div
    className="imageLine"
  >
    <div
      className="blackSide"
    />
    <div
      className="blackMiddle"
    />
    <div
      className="blackSide"
    />
  </div>
  <div
    className="imageLine"
  >
    <div
      className="blackSide"
    />
    <div
      className="blackMiddle"
    />
    <div
      className="blackSide"
    />
  </div>
  <div
    className="imageLine"
  >
    <div
      className="blackSide"
    />
    <div
      className="blackMiddle"
    />
    <div
      className="blackSide"
    />
  </div>
</div>
`;
